Sources/Subs-Attachments.php

```diff
@@ -1420,4 +1457,125 @@ function prepareAttachsByMsg($msgIDs)
 	}
 }
 
+/**
+ * Gets the dimensions of an SVG image (specifically, of its viewport).
+ *
+ * See https://www.w3.org/TR/SVG11/coords.html#IntrinsicSizing
+ *
+ * @param string $filepath The path to the SVG file.
+ * @return array The width and height of the SVG image in pixels.
+ */
+function getSvgSize($filepath)
+{
+	preg_match('/<svg\b[^>]*>/', file_get_contents($filepath, false, null, 0, 480), $matches);
+	$svg = $matches[0];
+
+	// If the SVG has width and height attributes, use those.
+	// If attribute is missing, SVG spec says the default is '100%'.
+	// If no unit is supplied, spec says unit defaults to px.
+	foreach (array('width', 'height') as $dimension)
+	{
+		if (preg_match("/\b$dimension\s*=\s*([\"'])\s*([\d.]+)([\D\S]*)\s*\\1/", $svg, $matches))
+		{
+			$$dimension = $matches[2];
+			$unit = !empty($matches[3]) ? $matches[3] : 'px';
+		}
+		else
+		{
+			$$dimension = 100;
+			$unit = '%';
+		}
+
+		// Resolve unit.
+		switch ($unit)
+		{
+			// Already pixels, so do nothing.
+			case 'px':
+				break;
+
+			// Points.
+			case 'pt':
+				$$dimension *= 0.75;
+				break;
+
+			// Picas.
+			case 'pc':
+				$$dimension *= 16;
+				break;
+
+			// Inches.
+			case 'in':
+				$$dimension *= 96;
+				break;
+
+			// Centimetres.
+			case 'cm':
+				$$dimension *= 37.8;
+				break;
+
+			// Millimetres.
+			case 'mm':
+				$$dimension *= 3.78;
+				break;
+
+			// Font height.
+			// Assume browser default of 1em = 1pc.
+			case 'em':
+				$$dimension *= 16;
+				break;
+
+			// Font x-height.
+			// Assume half of font height.
+			case 'ex':
+				$$dimension *= 8;
+				break;
+
+			// Font '0' character width.
+			// Assume a typical monospace font at 1em = 1pc.
+			case 'ch':
+				$$dimension *= 9.6;
+				break;
+
+			// Percentage.
+			// SVG spec says to use viewBox dimensions in this case.
+			default:
+				unset($$dimension);
+				break;
+		}
+	}
+
+	// Width and/or height is missing or a percentage, so try the viewBox attribute.
+	if ((!isset($width) || !isset($height)) && preg_match('/\bviewBox\s*=\s*(["\'])\s*[\d.]+[,\s]+[\d.]+[,\s]+([\d.]+)[,\s]+([\d.]+)\s*\\1/', $svg, $matches, PREG_UNMATCHED_AS_NULL))
```

Comment:
```suggestion
	if ((!isset($width) || !isset($height)) && preg_match('/\bviewBox\s*=\s*(["\'])\s*[\d.]+[,\s]+[\d.]+[,\s]+([\d.]+)[,\s]+([\d.]+)\s*\\1/', $svg, $matches))
```
This is, I believe, PHP 7.2+

Reply:
Derp. Yes, it is.

Reply:
...and I don't actually need it anyway.